main.c

```diff
@@ -432,6 +432,43 @@ init_headless(struct vkcube *vc)
                  NULL,
                  &b->image);
 
+   VkFormatProperties2 formatProperties = {
+      .sType = VK_STRUCTURE_TYPE_FORMAT_PROPERTIES_2,
+   };
+
+   vkGetPhysicalDeviceFormatProperties2(vc->physical_device,
+		                        vc->image_format,
+					&formatProperties);
+
+   VkFormatFeatureFlags requiredFeatures = 0;
+   requiredFeatures = formatProperties.formatProperties.linearTilingFeatures;
+
+   VkFormatFeatureFlags useFlags = 0;
```

Comment:
These are *feature* flags, not *use* flags. The naming is important, because use flags usually mean the flags from `VkImageUsageFlags`.

The flag checks can be simplified, made more readable, by eliminating the temporary variables. Like this:
```
if (!(format_props.linearTilingFeatures & VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT)) {
   fail("...\n");
}
```